src/Microsoft.CodeQuality.Analyzers/UnitTests/ApiDesignGuidelines/DoNotCatchGeneralExceptionTypesTests.cs

```diff
@@ -328,6 +326,100 @@ End Namespace
             GetCA1031BasicResultAt(10, 25, "TestMethod"));
         }
 
+        [Fact]
+        public void CSharp_NoDiagnostic_GenericExceptionRethrown()
+        {
+            VerifyCSharp(@"
+            using System;
+            using System.IO;
+
+            namespace TestNamespace
+            {
+                class TestClass
+                {
+                    public static void TestMethod()
+                    {
+                        try 
+                        {
+                            FileStream fileStream = new FileStream(""name"", FileMode.Create);
+                        }
+                        catch (Exception e)
+                        {
+                            throw e;
```

Comment:
❓ Why `throw e` instead of `throw`?

Reply:
This is an additional test case. The other case is already covered by `CSharp_Diagnostic_GeneralCatchWithRethrowFromSpecificCatch`.

While it might not make much sense to `throw e`, it's still worth ensuring that a diagnostic is not emitted for it.

Reply:
@sharwell any further concerns?

Reply:
Nope that answers my question thanks!